File: backend/src/main/java/com/jeju/jeju/domain/user/dto/SocialLoginResponse.java
package com.jeju.jeju.domain.user.dto;

public class SocialLoginResponse {

    private String accessToken;
    private String refreshToken;
    private String tokenType;
    private long expiresIn;
    private boolean isNewUser;

    public SocialLoginResponse() {}

    public SocialLoginResponse(String accessToken, String refreshToken, String tokenType,
                                long expiresIn, boolean isNewUser) {
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
        this.tokenType = tokenType;
        this.expiresIn = expiresIn;
        this.isNewUser = isNewUser;
    }

    public String getAccessToken()  { return accessToken; }
    public String getRefreshToken() { return refreshToken; }
    public String getTokenType()    { return tokenType; }
    public long getExpiresIn()      { return expiresIn; }
    public boolean isNewUser()      { return isNewUser; }
}

File: frontend/lib/features/auth/data/models/social_login_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'social_login_response.freezed.dart';
part 'social_login_response.g.dart';

@freezed
class SocialLoginResponse with _$SocialLoginResponse {
  const factory SocialLoginResponse({
    required String accessToken,
    required String refreshToken,
    required String tokenType,
    required int expiresIn,
    required bool isNewUser,
  }) = _SocialLoginResponse;

  factory SocialLoginResponse.fromJson(Map<String, dynamic> json) =>
      _$SocialLoginResponseFromJson(json);
}
